### PR TITLE
Reuse already created checksum collections on persist

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

 * Support to update the checksum collection from bytes, which might be received or read from the file already. This is useful, if we want to avoid re-reading files etc.
 * Support to persist snapshot with given checksum collection. 
 * collection of checksum is no longe recreated and directly written to the directory.

## Related issues

relate #14045 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SfvChecksumImpl.java [hunk 1]
```diff
@@ -122,6 +122,15 @@ public void updateFromFile(final Path filePath) throws IOException {
     checksums.put(fileName, checksum.getValue());
   }
 
+  @Override
+  public void updateFromBytes(final String fileName, final byte[] bytes) {
+    combinedChecksum.update(fileName.getBytes(UTF_8));
+    final Checksum checksum = new CRC32C();
```
Comment (npepinpe) [inline snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SfvChecksumImpl.java:128]: :wrench: You can create it once for the whole class, and just call `CRC32C#reset` before every usage. Very minor optimization :upside_down_face:

**Review by npepinpe — APPROVED** [review]
:rocket: I think we'll need one where you pass the checksum value itself (either as a long or as a byte array), and not the byte array of contents (since RocksDB will give us the checksum and not the file contents). It might also be useful if we try to replicate snapshot files in smaller chunks :shrug: 

Anyway, we'll cross that bridge when we get there :smile:

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ private static boolean isNotMetadataFile(final Path file) {
 
   public static void persist(final Path checksumPath, final ImmutableChecksumsSFV checksum)
       throws IOException {
-    try(final var stream = new FileOutputStream(checksumPath.toFile())) {
+    try (final var stream = new FileOutputStream(checksumPath.toFile())) {
```
Comment (npepinpe) [inline snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java:65]: ups :see_no_evil:

Comment (Zelldon) [reply snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java:65]: Yeah :D seems to be from the other PR :D

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SfvChecksumImpl.java (hunk 1 above)
Comment (Zelldon) [reply snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SfvChecksumImpl.java:128]: I think I keep it for now.

The CRC323C#reset :D 

```
    public void reset() {
        this.crc = -1;
    }
```

Comment (npepinpe) [reply snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SfvChecksumImpl.java:128]: If you look at how it works, this is the initial value :upside_down_face: Then when you pass things it gets updated. So it is, in fact, resetting the state to the initial values ;)

**Pull request merged**